File: WS_EUREKABANK_SOAPDOTNET_GR04/02.CLICON/EB_DOTNET_SOAP_CliCon/ec.edu.monster.controlador/CliCon_Controlador.cs
using EB_DOTNET_SOAP_CliCon.ServicioEB;
using System.Collections.Generic;
using System.Linq;

namespace EB_DOTNET_SOAP_CliCon.ec.edu.monster.controlador
{
    public class CliCon_Controlador
    {
        private readonly WSEurekaSoapClient servicio;

        public CliCon_Controlador()
        {
            // Fuerza el uso del endpoint "WSEurekaSoap" definido en App.config
            servicio = new WSEurekaSoapClient("WSEurekaSoap");
        }

        public string ValidarIngreso(string usuario, string password)
            => servicio.validarIngreso(usuario, password);

        // OJO: usa el tipo 'movimiento' del proxy
        public List<movimiento> ObtenerMovimientos(string cuenta)
        {
            var datos = servicio.traerMovimientos(cuenta);
            // Al regenerar, 'traerMovimientos' devuelve 'movimiento[]'
            return datos?.ToList() ?? new List<movimiento>();
        }

        public OperacionCuentaResponse RegistrarDeposito(string cuenta, decimal importe)
            => servicio.regDeposito(cuenta, importe);

        public OperacionCuentaResponse RegistrarRetiro(string cuenta, decimal importe)
            => servicio.regRetiro(cuenta, importe);

        public OperacionCuentaResponse RegistrarTransferencia(string cuentaOrigen, string cuentaDestino, decimal importe)
            => servicio.regTransferencia(cuentaOrigen, cuentaDestino, importe);
    }
}

File: WS_EUREKABANK_SOAPJAVA_GR04/02.CLICON/EurekaBank_SOAP_CliCon/src/ec/edu/monster/controlador/CliCon_Controlador.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.monster.controlador;

import ec.edu.monster.servicio.EurekaService;
import ec.edu.monster.ws.Movimiento;
import java.util.List;

/**
 *
 * @author JOIS
 */
public class CliCon_Controlador {
    
    public List<Movimiento> traerMovimientos(String cuenta){
        EurekaService service=new EurekaService();
       return service.traerMovimientos(cuenta);
           
    }
    
    public int regDeposito(String cuenta,double importe){
        EurekaService service=new EurekaService();
       return service.regDeposito(cuenta,importe);
           
    }
    
    public int regRetiro(String cuenta,double importe){
       EurekaService service=new EurekaService();
       return service.regRetiro(cuenta, importe);
    }
    
    public int regTransferencia(String cuentaOrigen, String cuentaDestino,double importe){
       EurekaService service=new EurekaService();
       return service.regTransferencia(cuentaOrigen, cuentaDestino, importe);
    }

    
    /*
    public String validarIngreso (String usuario, String password){
        EurekaService service=new EurekaService();
        return service.validarIngreso(usuario, password);
    }
    */
}